From 5c3e81e307e95cccefe0227273b819f240d4d040 Mon Sep 17 00:00:00 2001
From: Blueve <blueve@users.noreply.github.com>
Date: Mon, 26 Oct 2020 06:44:59 +0000
Subject: [PATCH 1/3] Put style as line number to ssh session environment
 variable

By default, the content between : and @ will be trimmed by sshd before it do
authentication and the trimmed string will be dropped silently. To use this
segment as line number for reverse SSH feature, we need to modify the source
code of OpenSSH and put this segment to a environment variable
SSH_TARGET_CONSOLE_LINE, then we can insert a short script into /etc/bash.bashrc
and run command consutil connect $SSH_TARGET_CONSOLE_LINE to enter the
management session automatically after user login.
---
 session.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/session.c b/session.c
index 492463529..768d3b3bb 100644
--- a/session.c
+++ b/session.c
@@ -1184,6 +1184,11 @@ do_setup_env(struct ssh *ssh, Session *s, const char *shell)
 		child_set_env(&env, &envsize, "SSH_ORIGINAL_COMMAND",
 		    original_command);
 
+	/* Take advantage of authentication style field */
+	if (s->authctxt->style)
+		child_set_env(&env, &envsize, "SSH_TARGET_CONSOLE_LINE",
+		    s->authctxt->style);
+
 	if (debug_flag) {
 		/* dump the environment */
 		fprintf(stderr, "Environment:\n");
-- 
2.43.0

